src/ctim/schemas/openc2_network.clj

```diff
@@ -0,0 +1,104 @@
+(ns ctim.schemas.network_coa
+  (:require [schema.core :as s]
+            [ctim.schemas.openc2vocabularies :as openc2v]))
+
+
+
+
+(s/defschema BGPBlackhole
+  {:type (s/enum "BGPBlackhole")
+   :host s/Str})
+
+(s/defschema DNSSinkhole
+  {:type (s/enum "DNSSinkhole")
+   :host s/Str})
+
+(def Protocol
+  (s/enum "TCP"
+          "UDP"
+          "ICMP"
+          "Any"))
+
+(def ACLAction
+  (s/enum "ALERT" 
+          "DROP"
+          "DENY"
+          "LOG"
+          "PASS"
+          "REJECT"))
+
+(s/defschema Traffic
+  {:protocol Protocol
+   :sourceAddress s/Str
+   :sourcePort s/Str
+   :destinationAddress s/Str
+   :destinationPort s/Str})
+
+(s/defschema NetworkACL
+  {:type (s/enum "NetworkACL")
+   :traffic Traffic
+   :action ACLAction})
+
+
+(s/def vlanProfile
+  {:vlanTag s/Str})
+
+(s/def secGroupProfile
+  {:secGroupTag s/Str
+   :secGroupACL s/Str})
+
+
+(s/defschema Remediation
+  {:type (s/enum "Remediation")
+   :server s/Str
+   :containmentProfile (s/either vlanProfile
+                                 secGroupProfile)
+   :acl NetworkACL})
+
+(s/defschema NonSensitive
+  {:type (s/enum "NonSensitive")
+   :permissibleIps [s/Str]
+   :acl NetworkACL})
+
+(s/defschema Honeypot
+  {:type (s/enum "Honeypot")
+   :permissibleIps [s/Str]
+   :acl NetworkACL 
+   :routes {:prefix s/Str
+            :NextHop s/Str
+            :NextHopType s/Str}})
+
+(def Encapsulation
+  (s/enum "GRE"
+          "VXLAN"))
+
+(s/defschema BlockModifier
+  {:type (s/enum 
+           "Perimeter"
+           "Internal")
+   :method (s/conditional
+            #(= "NetworkACL" (:type %)) NetworkACL
+            #(= "BGPBlackhole" (:type %)) BGPBlackhole
+            #(= "DNSSinkhole" (:type %)) DNSSinkhole)})
+
+
+  
+
+(s/defschema ContainModifier
+  {:type (s/eq "Contain")
+   :method (s/conditional
```

Comment:
Sure .... I'm new to Clojure so didn't know any better. Please recommend an alternative approach.

Thanks,
Jyoti

From: gbuisson <notifications@github.com<mailto:notifications@github.com>>
Reply-To: threatgrid/ctim <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Thursday, July 28, 2016 at 1:55 AM
To: threatgrid/ctim <ctim@noreply.github.com<mailto:ctim@noreply.github.com>>
Cc: Jyoti Verma <jyoverma@cisco.com<mailto:jyoverma@cisco.com>>, Author <author@noreply.github.com<mailto:author@noreply.github.com>>
Subject: Re: [threatgrid/ctim] issue-178: OpenC2 commands for network and SDN actuator types (#32)


In src/ctim/schemas/openc2_network.clj<https://github.com/threatgrid/ctim/pull/32#discussion_r72586396>:

> +
> +(s/defschema BlockModifier
> +  {:type (s/enum
> +           "Perimeter"
> +           "Internal")
> +   :method (s/conditional
> +            #(= "NetworkACL" (:type %)) NetworkACL
> +            #(= "BGPBlackhole" (:type %)) BGPBlackhole
> +            #(= "DNSSinkhole" (:type %)) DNSSinkhole)})
> +
> +
> +
> +
> +(s/defschema ContainModifier
> +  {:type (s/eq "Contain")
> +   :method (s/conditional


Can we use something else than conditional here?
It doesn't play well with swagger.

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/threatgrid/ctim/pull/32/files/9293ad93c7456e7e508c7827c6f1700462440e9c#r72586396>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFBR0gPxXxX6L3C_04Weiqyu0LKp6001ks5qaG6bgaJpZM4JJyo6>.